src/app/qgisapp.cpp

```diff
@@ -11154,7 +11156,22 @@ void QgisApp::toggleEditing()
 
 bool QgisApp::toggleEditing( QgsMapLayer *layer, bool allowCancel )
 {
-  QgsVectorLayer *vlayer = qobject_cast<QgsVectorLayer *>( layer );
+  switch ( layer->type() )
+  {
+    case QgsMapLayerType::VectorLayer:
+      return toggleEditingVectorLayer( qobject_cast<QgsVectorLayer *>( layer ), allowCancel );
+      break;
+    case QgsMapLayerType::MeshLayer:
+      return toggleEditingMeshLayer( qobject_cast<QgsMeshLayer *>( layer ) );
```

Comment:
we do not allow cancel?

Reply:
even if we do not support it ATM, I would add it to the toggleEditingMeshLayer as arg and just not use it ATM?

Reply:
we always allow cancel, that why we do not need this flag.

Reply:
but inddeed we have to not allow cancel and add it